scripts/examples/mixed_str_with_func/str_with_in.py

```diff
@@ -0,0 +1,7 @@
+"""Tests behaviour when string starts with 'log_file', ends with '.txt', and contains 'name' in between."""
+def str_with_in(param: str) -> bool:
+    if param.startswith("log_file"):
+        if param.endswith(".txt"):
+            if "name" in param[5:-5]:  
```

Comment:
Why the indexing? We are not handling indexing and the complexity of the test is not reduced if using:

`if "name" in param: `